src/zimscraperlib/image/optimization.py

```diff
@@ -280,21 +289,41 @@ def optimize_image(
                 "FMT": convert to format FMT (use Pillow names)"""
 
     src_format, dst_format = format_for(src, from_suffix=False), format_for(dst)
+
+    if src_format is None:  # pragma: no cover
+        # never supposed to happens since we get format from suffix, but good for type
+        # checker + code safety / clean errors
+        raise ValueError("Impossible to guess format from src image")
+    if dst_format is None:
+        raise ValueError("Impossible to guess format from dst image")
     # if requested, convert src to requested format into dst path
     if convert and src_format != dst_format:
         src_format = dst_format = convert if isinstance(convert, str) else dst_format
-        convert_image(src, dst, fmt=src_format)  # pyright: ignore
+        convert_image(src, dst, fmt=src_format)
         src_img = pathlib.Path(dst)
     else:
         src_img = pathlib.Path(src)
 
-    {  # pyright: ignore
-        "JPEG": optimize_jpeg,
-        "PNG": optimize_png,
-        "GIF": optimize_gif,
-        "WEBP": optimize_webp,
-    }.get(src_format)(src_img, dst, **options)
+    get_optimization_method(src_format)(src_img, dst, **options)
 
     # delete src image if requested
     if delete_src and src.exists() and src.resolve() != dst.resolve():
         src.unlink()
+
+
+def get_optimization_method(fmt: str):
+    """Return the proper optimization method to call for a given image format"""
```

Comment:
The reason to expose it is that:
- this method was already present in the tests
- this same code was duplicated in the `optimize_image` method

I thought it would help to make it public, since it might help some scraper ; I don't mind much if we decide this should a be a "private" method, but at least I don't want the exact same logic at two places in the code.

The fact it has no return type is a mistake